src/app/libraries/csrf_lib.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class csrf_lib extends Library {
+
+    public function __construct() {
+        if (session_status() == PHP_SESSION_NONE) {
+            session_start();
+        }
+    }
+    /**
+     * Generates a new CSRF token
+     * @return string Base64 encoded token
+     */
+    public static function getNewCSRFToken() {
+        return base64_encode(openssl_random_pseudo_bytes(12));
+    }
+
+    /**
+     * Generates a new CSRF token and sets it as the current token
+     * @return string The token
+     */
+    public function setNewCSRFToken() {
+        $csrfToken = self::getNewCSRFToken();
+        $_SESSION['csrf_token'] = $csrfToken;
+        return $csrfToken;
+    }
+
+    /**
+     * Checks whether CSRF is set and whether the current CSRF token from
+     * the POST data is correct. If not, returns a 400 HTTP response code,
+     * loads the 400 view and quits.
+     */
+    public function checkCSRFToken($value) {
+        if (! isset($_SESSION['csrf_token'])
+            || $value !== $_SESSION['csrf_token']) {
```

Comment:
I think it is common practice to put even the closing bracket of condition in new line.